eng/automation/generate.py

```diff
@@ -250,8 +252,22 @@ def sdk_automation_typespec(config: dict) -> List[dict]:
         tsp_projects = [tsp_projects]
 
     for tsp_project in tsp_projects:
+        # folder structure v2: specification/{service}/[data-plane|resource-manager]/{provider}/
+        folder_structure_v2_pattern = r"specification/.*/(data-plane|resource-manager)"
```

Comment:
@XiaofeiCao {service} folder in V2 is only one level. So the regex should be `specification/[^/]+/(data-plane|resource-manager)`

Reply:
Got it. This is only for TypeSpec, and in v1 we don't have `data-plane` or `resource-manager` segment for TypeSpec. I think it should be fine.